test/torch_test.py

```diff
@@ -488,8 +488,42 @@ def test_send_var_with_gradient(self):
         assert model.grad.id in remote._objects
         assert model.grad.data.id in remote._objects
 
+    def test_remote_optim_step(self):
+
+        torch.manual_seed(42)
+        hook = TorchHook(verbose=False)
+        local = hook.local_worker
+        local.verbose = False
+        remote = VirtualWorker(id=1, hook=hook, verbose=False)
+        local.add_worker(remote)
+        param = []
+
+        data = Var(torch.FloatTensor([[0, 0], [0, 1], [1, 0], [1, 1]])).send(remote)
+        target = Var(torch.FloatTensor([[0], [0], [1], [1]])).send(remote)
+
+        model = nn.Linear(2, 1)
+        opt = optim.SGD(params=model.parameters(), lr=0.1)
+
+        for i in model.parameters():
+            param.append(i[:])
+
+        model.send_(remote)
+        model.zero_grad()
+        pred = model(data)
+        loss = ((pred - target) ** 2).sum()
+        loss.backward()
+        opt.step()
+
+        model.get_()
+        for i in model.parameters():
+            param.append(i[:])
+
+        assert not ((param[0] == param[2]).all())
+        assert not ((param[1] == param[3]).all())
```

Comment:
Could these assertions be specific values as well as opposed to simply checking for equality. It could break in all sorts of terrible ways (param[2] could equal -inf, NaN, or None for example) and this test would still pass. Requiring it to equal a specific value, however, makes the test much more precise